### PR TITLE
[FLORA-132] Only index versionless package pages

### PR DESCRIPTION
**Pull request by RaoulHC**
Adds the robots noindex meta tag on package pages unless they're explicitly versioned.

## Contributor checklist

- [x] My PR is related to #132 
- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/flora-pm/flora-server/blob/development/CONTRIBUTING.md)
- [x] I have inserted my change and a link to this PR in the [CHANGELOG](https://github.com/flora-pm/flora-server/blob/development/CHANGELOG.md)

### CODE REVIEW

File: src/web/FloraWeb/Templates/Types.hs [hunk 1]
```diff
@@ -53,6 +53,7 @@ data TemplateEnv = TemplateEnv
   , environment :: DeploymentEnv
   , activeElements :: ActiveElements
   , assets :: Assets
+  , index :: Bool
```
Comment (tchoutri) [inline src/web/FloraWeb/Templates/Types.hs:56]: I would rather have this field named `indexPage` :)

File: src/web/FloraWeb/Components/Header.hs [hunk 2]
```diff
@@ -33,6 +33,7 @@ header = do
       head_ $! do
         meta_ [charset_ "UTF-8"]
         meta_ [name_ "viewport", content_ "width=device-width, initial-scale=1"]
+        unless index $ meta_ [name_ "robots", content_ "noindex"]
```
Comment (tchoutri) [inline src/web/FloraWeb/Components/Header.hs:36]: Excellent, thank you :)

**Review by tchoutri — CHANGES_REQUESTED** [review]

**Comment by tchoutri** [comment]
@mergifyIO rebase

**Comment by RaoulHC** [comment]
I thought the PR title check might have been flakey, but on second look looks like it's trying to pull my branch, but not from the fork so is having a sad time. I'm not too familiar with github workflows but do we want to get rid of the with github.head_ref bit?

**Comment by tchoutri** [comment]
> I thought the PR title check might have been flakey, but on second look looks like it's trying to pull my branch, but not from the fork so is having a sad time. I'm not too familiar with github workflows but do we want to get rid of the with github.head_ref bit?

@RaoulHC yep, definitely :)

**Review by tchoutri — APPROVED** [review]

**Pull request merged**